hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieInternalConfig.java

```diff
@@ -46,6 +46,13 @@ public class HoodieInternalConfig extends HoodieConfig {
       .withDocumentation("For SQL operations, if enables bulk_insert operation, "
           + "this configure will take effect to decide overwrite whole table or partitions specified");
 
+  public static final ConfigProperty<Boolean> ENABLE_PREPPED_MERGE_WRITES = ConfigProperty
+      .key("hoodie.internal.sql.prepped.merge.enabled")
+      .defaultValue(true)
+      .markAdvanced()
+      .sinceVersion("0.14.0")
+      .withDocumentation("Controls whether spark sql optimized merge is enabled.");
```

Comment:
Lets move this to HoodieWriteConfig and ignore the one we are adding in DataSourcEwriteOptions

Reply:
lets name the the config 
hoodie.spark.sql.optimized.merge.enable

Reply:
Fixed.